### PR TITLE
6.x: rename _accessible to patchable

### LINKED ISSUE #17375: 6.0: Rename `EntityTrait::_accessible` property to something more appropriate
### Description

I never really liked the name of the `_accessible` property inside our `Cake\Datasource\EntityTrait` since just by its name it indicates some sort of permission system to prevent (general) access to specified properties inside PHP... which it doesn't do.

The concept of `mass assignment` is properly described in the [Docs](https://book.cakephp.org/4/en/orm/entities.html#mass-assignment) and the [baked entity classes](https://github.com/cakephp/bake/blob/3.x/templates/bake/Model/entity.twig#L43-L52) also have a somewhat suitable explenation of what is happening but still the property name itself is a bit missleading imho.

Here are some of my (bad) suggestions that came to my mind:
* `massAssignable` (to be right on the nose but still what does `assignable` mean)
* `massEditable` (although this also doesn't represent it correctly because `$entity->set('myfield', 'value');` would also be blocked if `myfield` is set to `false`)

Would love to hear your thoughts on that one 😁 

### CakePHP Version

6.0

### PR DESCRIPTION
**Pull request by LordSimal**
Resolves #17375 

The following Rector configs need to be created to help users upgrade to this new name:

* Rename EntityInterface methods
  - `setAccess` => `setPatchable`
  - `getAccessible` => `getPatchable`
  - `isAccessible` => `isPatchable`
* Rename EntityTrait property `_accessible` => `patchable`
* Rename Marshaller/ORM method option `accessibleFields` => `patchableFields`

### CODE REVIEW

File: src/ORM/phpstan.neon.dist [hunk 1]
```diff
@@ -24,3 +24,4 @@ parameters:
 		- '#^Parameter \#1 \$fields of method Cake\\Datasource\\EntityInterface::.*\(\) expects list<string>, array<string> given\.$#'
 		- '#^Parameter \#1 \$field of method Cake\\Datasource\\EntityInterface::has\(\) expects list<string>\|string, .+<string> given\.$#'
 		- '#^Parameter \#1 \$field of method Cake\\Datasource\\EntityInterface::unset\(\) expects list<string>\|string, array<string>\|string given\.$#'
+		- '#^Call to an undefined method Cake\\Datasource\\EntityInterface::setPatchable\(\)\.$#'
```
Comment (LordSimal) [inline src/ORM/phpstan.neon.dist]: I have no idea why PHPStan now doesn't understand that method exists on the EntityInterface even though it had no problem previously

File: src/ORM/Marshaller.php [hunk 2]
```diff
@@ -147,8 +147,8 @@ protected function _buildPropertyMap(array $data, array $options): array
      *   Defaults to true/default.
      * - associated: Associations listed here will be marshalled as well. Defaults to null.
      * - fields: An allowed list of fields to be assigned to the entity. If not present,
-     *   the accessible fields list in the entity will be used. Defaults to null.
-     * - accessibleFields: A list of fields to allow or deny in entity accessible fields. Defaults to null
+     *   the patchable fields list in the entity will be used. Defaults to null.
+     * - patchableFields: A list of fields to allow or deny in entity patchable fields. Defaults to null
```
Comment (ADmad) [inline src/ORM/Marshaller.php:151]: Wondering if we could shorten this to just `patchable`. The `Fields` suffix seems unnecessary.

Comment (dereuromark) [reply src/ORM/Marshaller.php:151]: patchable does sound a bit like a boolean flag, though.
But I can understand that the length might be a bit cumbersome.

**Review by markstory — COMMENTED** [review]
I like the new method names and property names. My only concern is the amount of toil this will add to the 6.x upgrade. We'll need to make sure folks are aware of the upgrade tooling more than in the past :smile:

**Comment by LordSimal** [comment]
> We'll need to make sure folks are aware of the upgrade tooling more than in the past 😄

Then I will make a Youtube Video showcasing the Upgrade tool once again (since no one seems to know I showcased it in CakeFest 2023)

Written docs are already present and maybe need a few touch ups in the end.

**Comment by ADmad** [comment]
The change in methods name would be easy to spot if one misses the rename as it would just generate an error. The array key change would be silently ignored if missed. So for those we should keep a check and show a warning.

File: src/ORM/phpstan.neon.dist (hunk 1 above)
Comment (LordSimal) [reply src/ORM/phpstan.neon.dist]: Rebasing solved this. Seems this was an issue with an earlier phpstan version

**Comment by dereuromark** [comment]
Looks good :+1:
